AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java

```diff
@@ -1247,13 +1247,15 @@ public boolean undoAvailable() {
 
 
     public long undo() {
-    	Object[] data = mUndo.removeLast();
-        Timber.d("undo() of type %s", data[0]);
-    	switch ((DismissType) data[0]) {
+        Pair<DismissType, Object[]> lastUndo = mUndo.removeLast();
+        DismissType dt = lastUndo.first;
+    	Object[] data = lastUndo.second;
+        Timber.d("undo() of type %s", dt);
+    	switch (dt) {
```

Comment:
```suggestion
        Pair<DismissType, Object[]> lastUndo = mUndo.removeLast();
        DismissType dt = lastUndo.first;
        Object[] data = lastUndo.second;
        Timber.d("undo() of type %s", dt);
    	switch (dt) {
```

Just a whitespace thing - it was a tab instead of spaces